Tests/FluentAssertions.Specs/TypeDescriptionUtilitySpecs.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+using Xunit;
+
+namespace FluentAssertions.Specs
+{
+    public class TypeDescriptionUtilitySpecs
```

Comment:
🔧 We don't need these tests since it is an implementation detail of the new assertions and all possible paths should already be covered by their tests.

Reply:
The existing tests of the new assertions don't exercise every possible way in which a `Type` can be described, and they also don't currently spell out exactly how the types should be described. I could add tests/test cases to do this further upstream, but doesn't that violate a core principle of "unit" testing??

Reply:
Although you're right in general, supporting code is always part of the implementation of the APIs, and in general is not tested separately. By the way, "unit" doesn't mean "class". That's a common misconception. If we would practice that, we would be testing every little class that is hidden behind the `BeEquivalentTo` method separately.

Reply:
Okay, but this utility class _is_ basically a unit, isn't it? Doesn't it decrease the quality of the tests to _only_ test it indirectly via one of its consumers??

Reply:
Looking at the output, it does look like a nice addition that could be promoted to become a dedicated implementation of `IValueFormatter`. But maybe it's better to completely move this `TypeDescriptorUtility` into a new PR and make it a value formatter. I expect it to affect a lot of tests. It'll also avoid this discussion and allow you to move the PR forward. What do you think @jnyrup ?

Reply:
My two cents are that it's useful in its current form and is easily refactorable in a subsequent PR -- so instead of removing it from this PR to make a new one, why not _just_ make a new one that changes its form to the desired one? It serves a purpose in this PR as it is.

Reply:
No, since it then creates an inconsistency, just remove it altogether from the PR.

Reply:
So how to handle the fact that this functionality is used from 3 different places?? Do I copy/paste the functionality into each of those places??

Reply:
It shouldn't be used at all. Just rely on the formatting logic to create a display value for the object.

Reply:
Hmm, okay. The thing that led to me creating the class in the first place was the fact that if people use LINQ, then the object's type is going to be _completely and utterly meaningless_.

Reply:
Yes, and it's a very nice way of updating the default formatter. So I would most definitely welcome a separate PR for that.

Reply:
Okeydoke :-)